core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -444,9 +450,31 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
       FileSystemContext.INSTANCE.reset();
       LineageContext.INSTANCE.reset();
 
-      sFileSystem = FileSystem.Factory.get();
-      sInitialized = true;
+      // Try to connect to master, if it fails, the provided uri is invalid.
+      FileSystemMasterClient client = FileSystemContext.INSTANCE.acquireMasterClient();
+      try {
+        client.connect();
+        // Connected, initialize.
+        sFileSystem = FileSystem.Factory.get();
+        sInitialized = true;
+      } catch (ConnectionFailedException | IOException e) {
+        LOG.error("Failed to connect to the provided master address {}: {}.", uri, e);
+        throw new IOException(e);
+      } finally {
+        FileSystemContext.INSTANCE.releaseMasterClient(client);
+      }
+    }
+  }
+
+  private void assertSameClientURI() throws IOException {
```

Comment:
Should `assertSameClientURI` also be used in the other interfaces (ie. Alluxio FileSystem)?